### PR TITLE
Fixes #22072 - adds scrolling in bookmarks dropdown

### PR DESCRIPTION
**Pull request by amirfefer**
before:
hosts 1

after:
hosts

### CODE REVIEW

**Comment by timogoebel** [comment]
Just a suggestion: Why not limit the scrollable area to the actual bookmark list and always show the "Bookmark this search" and "Documentation" links?

File: app/assets/stylesheets/base.scss [hunk 1]
```diff
@@ -655,3 +655,9 @@ code.transparent {
   padding: 0;
   margin-bottom: 0;
 }
+
+.scrollable-dropdown {
+  max-height: 400px;
+  width: 175px;
+  overflow-y: auto;
```
[the remark this answers is not in the document]
Comment (sharvit) [reply app/assets/stylesheets/base.scss]: @amirfefer can you add an empty line at the end of the file? :smiley:

File: webpack/assets/javascripts/react_app/components/bookmarks/index.js [hunk 2]
```diff
@@ -39,7 +39,14 @@ class BookmarkContainer extends React.Component {
     ) : (
       <Dropdown pullRight id={controller}>
         <Dropdown.Toggle />
-        <Dropdown.Menu>
+        <Dropdown.Menu className='scrollable-dropdown'>
+          {canCreate && (
+            <MenuItem key="newBookmark" id="newBookmark" onClick={this.onClick.bind(this)}>
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/bookmarks/index.js]: `onClick` should be bind at the constructor with the help of the helpers.

I prefer naming those event as well:
```js
onClick={this.handleNewBookmarkClick}
```

File: webpack/assets/javascripts/react_app/components/bookmarks/index.js [hunk 3]
```diff
@@ -39,7 +39,14 @@ class BookmarkContainer extends React.Component {
...
+              {__('Bookmark this search')}
+            </MenuItem>
+          )}
+          <DocumentationUrl id="bookmarkDocumentation" href={this.props.documentationUrl} />
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/bookmarks/index.js]: `documentationUrl` can be extracted from props at the beginning of this function.

**Review by sharvit — CHANGES_REQUESTED** [review]

**Comment by amirfefer** [comment]
@sharvit Thanks, would you have another look?

**Review by sharvit — CHANGES_REQUESTED** [review]
@amirfefer thanks, it looks good!

Can you just add an empty line at the end of `base.scss`?

hound says: `Unexpected missing end-of-source newline no-missing-end-of-source-newline`

**Comment by sharvit** [comment]
@timogoebel I think it can be nice to try, but i afraid it may feel unnatural becuase the scrollbar will be shorter and under the top controllers.

It won't be a small twik since we will need add shadows effects between the top controllers and the list so it will feel like the list is getting under the top controllers.
It mostly mean to break the patternflty design, i would feel much more comfortable about it if paternfly would support such a design.

@Rohoover?

**Comment by Rohoover** [comment]
I think adding the scroll is an improvement.

I also think if there is time and it's not too much effort @timogoebel suggestion would be nice as well. I don't think having a floating header/footer is particularly unnatural.

**Comment by Rohoover** [comment]
Hey also wanted to post some concept work I've done with the filter/bookmark bar for thought:

search_v2

**Comment by ohadlevy** [comment]
thanks! what does the "x" next to the bookmark name mean? does it delete it?

**Comment by Rohoover** [comment]
@ohadlevy 

Yes, I was looking at ways to better remove bookmarks.

File: app/assets/stylesheets/base.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (amirfefer) [reply app/assets/stylesheets/base.scss]: I already added, as you can see hound has passed.

**Comment by amirfefer** [comment]
I've tried to add floating dropdown items, however bootstrap's dropdown  makes it harder to implement, and I don't think it worth the time.
I've added a `bookmark` icon though.  

The item's delete action is not related to scrolling,  it looks great though.  I prefer to add it on different PR

hosts 2

**Comment by timogoebel** [comment]
Looks good to me.

I also like the "inline removal" of the bookmarks.

@Rohoover: Can you please create a Remine issue for that and describe in more details how it should behave (e.g. does it need a "Confirm?" dialog etc.) and attach the concept work?

@sharvit: Do you have more comments? If you approve as well, I'll be happy to merge.

**Comment by sharvit** [comment]
@timogoebel Looks good to me but travis is faling.

I will re-review once the tests will pass.

**Comment by Rohoover** [comment]
@timogoebel Will do, thanks!

**Comment by waldenraines** [comment]
@amirfefer this needs a rebase please.

**Comment by amirfefer** [comment]
@sharvit , @waldenraines 
I've rebased, could you have another look?

**Review by waldenraines — APPROVED** [review]
Tested this out and it works well.  Code looks fine to me.  A definite improvement on the current behavior.

**Review by sharvit — APPROVED** [review]
Thanks @amirfefer!
I have tested it and it works great.

@amirfefer can you donate the class to `patternfly`?

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Merged, thanks @amirfefer. And thanks for the help in reviewing this @waldenraines and @sharvit.

I'm putting this on 1.17, cherry-picked as 9c06ba937d808e29c86047190776fdb38da274b1.

**Pull request merged**